### PR TITLE
update to vue2

### PR DESCRIPTION
**Pull request by krissrex**
Many changes, I recommend you test to make sure it works first. It is running fine for me, however.
More about migrations from vue1 to vue2 here: https://vuejs.org/v2/guide/migration.html

### Major changes:

* app component is wrapped in a div, because vue2 forbids `el: "body"`
* vue2, vuex2, and vue-loader 10
* alias for vue in webpack to use standalone version (needed for template compiler)
* `vuex` property on components is no longer used in vuex2. Instead, they are moved to `computed` and `methods`. TODO: use a cleaner syntax, instead of `this.$store`
* `ready` lifecycle event is now `mounted`. Note that you are not guaranteed to be "in document" in mounted, so `Vue.nextTick` or `this.$nextTick` might be needed.
* `v-el` is replaced with `ref=`
* components need `.native` for DOM events
* vuex mutations use `commit` instead of `dispatch`. Only actions use `dispatch` now.
* `$index` in `v-for` is now gone, so we must use `v-for="(tab, index) in tabs"` to get the index
* `{{{  }}}` interpolation for html is replaced with `v-html`

#### ~Preview code fonts are disabled~ Edit: fixed

I had to disable the `style` element in `src/components/main.vue`:

```
        <!--style>
          .markdown-body code,.markdown-body pre {
            font-family: {{ settings.preview.codeFont === 'inherit' ? settings.preview.font : settings.preview.codeFont }}
          }
        </style-->
```

The template compiler complains about styles not being allowed inside templates. That means custom preview code font is not working now.

#### transitions API changed

The file `src/utils/transitions.js` use the old `Vue.transition` API. That has been deprecated in favor of `transition` tags and custom components: https://vuejs.org/v2/guide/transitions.html#Reusable-Transitions .

I did not find any usages of it, so I didn't migrate it.

### CODE REVIEW

**Comment by krissrex** [comment]
This mixin might solve the `style in template` issue: https://github.com/paulpflug/vue-mixins#dynamiccss

**Comment by egoist** [comment]
bug-eme

There's a bug when we use Vue 2, if we close the tab on the left, the current tab content get changed too.

**Comment by krissrex** [comment]
> There's a bug when we use Vue 2, if we close the tab on the left, the current tab content gets changed too.

Nice catch. I'll look into it. 

To fix the editor code-font, I believe we can use `document.styleSheets` to manipulate a `style`-tag that we put in the head. Alternatively disable lint for it. I will investigate that later.

**Comment by krissrex** [comment]
>>There's a bug when we use Vue 2, if we close the tab on the left, the current tab content gets changed too.
>
> Nice catch. I'll look into it.

Hm, the DOM keeps the CodeMirror-element when you delete a tab, and instead updates the old div with data needed to display the tab. 

Given this case: 
* tab0 `{ content: "test0", editor: editor0 }`, CodeMirror `value: test0` from TextArea `editor-0`  bound to `Editor 0` and `tab0.content`0
* tab1 `{ content: "test1", editor: editor1] }`, CodeMirror `value: test1` from TextArea `editor-1`  bound to `Editor 1` and `tab1.content`

When deleting tab0, you end up with:
* tab1 `{ content: "test1", editor: editor1 }`, CodeMirror `value: test0` from TextArea `editor-0` bound to Editor 0  and `tab1.content`

Then:
`editor0.setContent("testBug")` will update the CodeMirror value for tab1, and `tab1.content` will be "testBug".

What I expected to happen is that the element from tab0 gets deleted, instead of reused. Since it is inserted on tab-creation, I must find a way to re-bind the CodeMirror DOM element, or replace it.

File: src/components/main.vue [hunk 1]
```diff
@@ -174,6 +175,13 @@
 
   const currentWindow = remote.getCurrentWindow()
   const config = currentWindow.$config
+  
+  const tabId = {
```
Comment (krissrex) [inline src/components/main.vue]: Would you prefer if I put this inside `utils/tab.js`?

Comment (egoist) [inline src/components/main.vue]: Yep, that sounds better.

**Comment by krissrex** [comment]
There's a bug in `main.vue`  when drag'n'dropping multiple files. 
Somehow `tab.editor` is null for all files that were dragged in except the first.

**Comment by krissrex** [comment]
This branch should be ready to merge now. Any more thoughts about it?

**Comment by egoist** [comment]
on your branch open the app with `npm start` but it cannot create a new tab

**Comment by krissrex** [comment]
Strange, it opens with a tab for me. Did you run `npm install` first? Any errors in the js-console?

**Comment by egoist** [comment]
After reinstall it works fine!

**Pull request merged**